### DISCUSSION TITLE
Error in page when using MenuIcon from MUI

### DISCUSSION THREAD

**Discussion by teukuridho** [vikejs/vike#1562] [Discussion] [answered]
### Description

Hi, I got this error when using `MenuIcon` from `MUI`
```
Warning: React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: object.

Check your code at +Page.tsx:4.
    at TestPage
    at div
    at IndexLayout (/Users/teukuridho/Documents/Works/PersonalWorks/vike-unilever-ert/Layouts/IndexLayout.tsx:5:3)
```

This is my `pages/test/+Page.ts`
```
import MenuIcon from '@mui/icons-material/Menu';
export default function TestPage() {
    return <div>
        <MenuIcon/>
    </div>
}
```

This is my `pages/+config.ts`

```
import { Config } from "vike/types";
import vikeReact from 'vike-react/config'
import Layout from "../Layouts/IndexLayout";
import Head from "../Heads/DefaultHead";

export default {
    Layout,
    Head,
    title: 'ERT',
    extends: vikeReact
} satisfies Config
```

It runs without error if I set `server: false, client: true` in meta
```
Page: {
    env: {
        server: false,
        client: true
    }
}
```

How do I resolve this without setting the mta? thanks

**UPDATE** :It runs when I wrap the `MenuIcon` with `ClientOnly`, like this
```
<ClientOnly 
    load={async() => MenuIcon}
    fallback={<p>Loading</p>}
>
    {(MenuIconComponent) => <MenuIconComponent sx={{
        color:"#4776E7",
        fontSize: 40
    }}/>}
</ClientOnly>
```
maybe this could be the temporary solution

also my another component, `Sidebar` that needs `window` got an error `window is not defined`, after wrapping it with `ClientOnly`, it runs; maybe there could be an easy way to use client component, like `'use client'`

**Comment by brillout** [comment, marked as answer]
Did you get the following hint?

```
┌────────────────────────────────────────────────────────────────────────────────────┐
│ To fix this error, see https://vike.dev/broken-npm-package#react-invalid-component │
└────────────────────────────────────────────────────────────────────────────────────┘
```

**Reply by teukuridho** [reply]
oh I missed it, with `cjsInterop` it works now, no need ClientOnly. thanks

**Reply by teukuridho** [reply]
sorry if this is off topic, but how do I refresh vite when express-entry changes when scaffolded with Bati?

**Reply by brillout** [reply]
https://brillout.github.io/rules/

**Reply by brillout** [reply]
Most notably https://brillout.github.io/rules/#try.

**Reply by teukuridho** [reply]
i have tried twith `tsx watch` but it kept refreshing even when I don't make any changes. I might have seen it somewhere in vike.dev but I forgot to save and can't find it now

**Reply by teukuridho** [reply]
turns out what causes the `tsx` keeps refreshing is vite making timestamp file, tried with `tsx watch ./express-entry.ts --ignore ./vite.config.ts* ` it seems to ignore the pattern
I literally spending atleast one hour here XD
also tried nodemon gives error `Unknown file extension ".ts"` tried with install `ts-node` samre result

**Reply by teukuridho** [reply]
well I found it on `server/index.ts` vike template app , with `vite-node`. thanks anyway
